src/app.cpp

```diff
@@ -86,6 +86,7 @@ void transform(
 				struct canfd_frame can = { 0 };
 				can.can_id = ntoh32(*((uint32_t*)data));
 				can.len = data[4];
+				can.flags |= header.data_flags & 0x02 ? CANFD_ESI;
```

Comment:
This line is for both TECMP_DATA_CAN and TECMP_DATA_CANFD

* bit 1 is ESI for TECMP_DATA_CANFD => good
* bit 1 is RTR for TECMP_DATA_CAN => not good.

Reply:
@aamereller you didn't get what I mean...